test/test_transforms_v2_refactored.py

```diff
@@ -2208,9 +2180,96 @@ def new_resize(dp, *args, **kwargs):
+
+        with pytest.raises(ValueError, match="Kernels can only be registered on dispatchers"):
+            F.register_kernel(datapoints.Image, F.resize)
+
+        with pytest.raises(ValueError, match="Kernels can only be registered for subclasses"):
+            F.register_kernel(F.resize, object)
+
+        F.register_kernel(F.resize, datapoints.Image)(F.resize_image_tensor)
+
+        with pytest.raises(ValueError, match="already has a kernel registered for type"):
+            F.register_kernel(F.resize, datapoints.Image)(F.resize_image_tensor)
+
+
+class TestGetKernel:
+    def make_and_register_kernel(self, dispatcher, input_type):
+        return _register_kernel_internal(dispatcher, input_type, datapoint_wrapper=False)(object())
+
+    @pytest.fixture
+    def dispatcher_and_kernels(self, mocker):
+        mocker.patch.dict(_KERNEL_REGISTRY, clear=True)
+
+        dispatcher = object()
+
+        kernels = {
+            cls: self.make_and_register_kernel(dispatcher, cls)
+            for cls in [
+                torch.Tensor,
+                PIL.Image.Image,
+                datapoints.Image,
+                datapoints.BoundingBoxes,
+                datapoints.Mask,
+                datapoints.Video,
+            ]
+        }
+
+        yield dispatcher, kernels
+
+    def test_unsupported_types(self, dispatcher_and_kernels):
+        dispatcher, _ = dispatcher_and_kernels
+
+        class MyTensor(torch.Tensor):
+            pass
+
+        class MyPILImage(PIL.Image.Image):
+            pass
+
+        for input_type in [str, int, object, MyTensor, MyPILImage]:
+            with pytest.raises(TypeError, match=re.escape(str(input_type))):
+                _get_kernel(dispatcher, input_type)
+
+    def test_exact_match(self, dispatcher_and_kernels):
+        dispatcher, kernels = dispatcher_and_kernels
+
+        for input_type, kernel in kernels.items():
+            assert _get_kernel(dispatcher, input_type) is kernel
+
+    def test_builtin_datapoint_subclass(self, dispatcher_and_kernels):
+        dispatcher, kernels = dispatcher_and_kernels
+
+        class MyImage(datapoints.Image):
+            pass
+
+        class MyBoundingBoxes(datapoints.BoundingBoxes):
+            pass
+
+        class MyMask(datapoints.Mask):
+            pass
+
+        class MyVideo(datapoints.Video):
+            pass
+
+        assert _get_kernel(dispatcher, MyImage) is kernels[datapoints.Image]
+        assert _get_kernel(dispatcher, MyBoundingBoxes) is kernels[datapoints.BoundingBoxes]
+        assert _get_kernel(dispatcher, MyMask) is kernels[datapoints.Mask]
+        assert _get_kernel(dispatcher, MyVideo) is kernels[datapoints.Video]
+
+    def test_datapoint_subclass(self, dispatcher_and_kernels):
+        dispatcher, _ = dispatcher_and_kernels
+
+        class MyDatapoint(datapoints.Datapoint):
+            pass
+
+        # Note that this will be an error in the future
+        assert _get_kernel(dispatcher, MyDatapoint) is _noop
```

Comment:
Follow up on comment about using `object` for the dispatcher:

Reading that kind of line typically makes me wonder what we're actually testing here. What real-world scenario does this correspond to? It's hard to wrap my head around these tests.